### ISSUE TITLE
Removing a bucket with significant number of files fails in v1.7.7

### ISSUE THREAD

**Issue by zerkms** [rook/rook#9555]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

**Expected behavior:**

**How to reproduce it (minimal and precise):**

Preamble: I'm running rook 1.7.7 and understand that it's not the latest, so if it's known and/or was fixed already - my apologies.

I have found that if a bucket removal is not instant and takes time - then rook does not handle timeout properly and then tries to reconcile already missing bucket.

With logs it will be more obvious:

```
I0111 01:50:52.988860 7 controller.go:218] "msg"="reconciling claim" "key"="redacted/redacted"
I0111 01:50:52.988964 7 helpers.go:107] "msg"="getting claim for key" "key"="redacted/redacted"
I0111 01:50:53.007913 7 helpers.go:213] "msg"="getting ObjectBucketClaim's StorageClass" "key"="redacted/redacted"
I0111 01:50:53.044281 7 helpers.go:218] "msg"="got StorageClass" "key"="redacted/redacted" "name"="objectstore-redacted"
I0111 01:50:53.044324 7 controller.go:246] "msg"="OBC deleted, proceeding with cleanup" "key"="redacted/redacted"
I0111 01:50:53.044332 7 controller.go:509] "msg"="syncing obc deletion" "key"="redacted/redacted"
I0111 01:50:53.045211 7 controller.go:640] "msg"="getting objectBucket for key" "key"="redacted/redacted"
I0111 01:50:53.072208 7 helpers.go:141] "msg"="getting configMap for key" "key"="redacted/redacted"
I0111 01:50:53.090822 7 helpers.go:154] "msg"="getting secret for key" "key"="redacted/redacted"
I0111 01:50:53.105496 7 resourcehandlers.go:463] "msg"="updating status:" "key"="redacted/redacted" "new status"="Released" "ob"="obc-redacted-redacted" "old status"="Bound"
I0111 01:50:53.138725 7 helpers.go:229] "msg"="getting ObjectBucket's storage class" "key"="redacted/redacted" "name"="objectstore-redacted"
I0111 01:50:53.150182 7 helpers.go:234] "msg"="got StorageClass" "key"="redacted/redacted" "name"=null
2022-01-11 01:50:53.152145 I | op-bucket-prov: getting storage class "objectstore-redacted"
2022-01-11 01:50:54.084321 I | op-bucket-prov: Delete: deleting bucket "redacted" for OB "obc-redacted-redacted"
2022-01-11 01:50:54.084508 I | op-bucket-prov: deleting Ceph user "ceph-user-1Ma8hVVg" and bucket "redacted"
E0111 01:51:09.087826 7 controller.go:205] error syncing 'redacted/redacted': provisioner error deleting bucket failed to delete OBCResource bucket "redacted": failed to delete bucket "redacted": Delete "http://rook-ceph-rgw-redacted.rook-ceph.svc:80/admin/bucket?bucket=redacted&format=json&purge-objects=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), requeuing
I0111 01:51:09.097656 7 controller.go:218] "msg"="reconciling claim" "key"="redacted/redacted"
I0111 01:51:09.097722 7 helpers.go:107] "msg"="getting claim for key" "key"="redacted/redacted"
I0111 01:51:09.104604 7 helpers.go:213] "msg"="getting ObjectBucketClaim's StorageClass" "key"="redacted/redacted"
I0111 01:51:09.112681 7 helpers.go:218] "msg"="got StorageClass" "key"="redacted/redacted" "name"="objectstore-redacted"
I0111 01:51:09.112862 7 controller.go:246] "msg"="OBC deleted, proceeding with cleanup" "key"="redacted/redacted"
I0111 01:51:09.113308 7 controller.go:509] "msg"="syncing obc deletion" "key"="redacted/redacted"
I0111 01:51:09.113389 7 controller.go:640] "msg"="getting objectBucket for key" "key"="redacted/redacted"
I0111 01:51:09.118661 7 helpers.go:141] "msg"="getting configMap for key" "key"="redacted/redacted"
I0111 01:51:09.123147 7 helpers.go:154] "msg"="getting secret for key" "key"="redacted/redacted"
I0111 01:51:09.127229 7 resourcehandlers.go:463] "msg"="updating status:" "key"="redacted/redacted" "new status"="Released" "ob"="obc-redacted-redacted" "old status"="Released"
I0111 01:51:09.142713 7 helpers.go:229] "msg"="getting ObjectBucket's storage class" "key"="redacted/redacted" "name"="objectstore-redacted"
I0111 01:51:09.151283 7 helpers.go:234] "msg"="got StorageClass" "key"="redacted/redacted" "name"=null
2022-01-11 01:51:09.151304 I | op-bucket-prov: getting storage class "objectstore-redacted"
2022-01-11 01:51:09.592141 I | op-bucket-prov: Delete: deleting bucket "redacted" for OB "obc-redacted-redacted"
2022-01-11 01:51:09.592167 I | op-bucket-prov: deleting Ceph user "ceph-user-1Ma8hVVg" and bucket "redacted"
E0111 01:51:13.686515 7 controller.go:205] error syncing 'redacted/redacted': provisioner error deleting bucket failed to delete OBCResource bucket "redacted": failed to delete bucket "redacted": UnknownError tx0000000000000000b9878-0061dce28d-54929ef-redacted 54929ef-redacted-redacted, requeuing
```

The bucket contains almost several thousands files and its removal is not instant. And we can see that after the removal api request was timed out - reconciliation failed. And it goes on and on and on infinitely (until I manually removed a finaliser from the OBC). And for the record: I did check manually with rgw the bucket was removed and also I manually removed a `ceph-user-1Ma8hVVg`.

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): ubuntu focal
* Kernel (e.g. `uname -a`): `5.4.0-89-generic #100-Ubuntu SMP Fri Sep 24 14:50:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud provider or hardware configuration: bare metal
* Rook version (use `rook version` inside of a Rook Pod): rook: v1.7.7
* Storage backend version (e.g. for ceph do `ceph -v`): ceph version 16.2.6 (ee28fb57e47e9f88813e24bbf4c14496ca299d31) pacific (stable)
* Kubernetes version (use `kubectl version`):  `Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.6"`
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)): HEALTH_OK

**Comment by thotz** [comment]
The bucket removal takes time, since it needs to delete all the contents. What was the size of the bucket and the number of objects in the bucket? So that we can work on something to improve the timeouts or something. Can you check rgw pods logs why delete was failing for the bucket

**Comment by zerkms** [comment]
It was a timeout, it was around 6Gb and 1k of files.

Does it really need to be timeouts tuning or instead it just should reconcile it in some other way? Eg: if it requested for deletion, but the deletion timed out - then wait til it disappears.

**Comment by thotz** [comment]
The error was 125, the operation was cancelled from the RGW, but I am wondering why what happens in the next deletion for reconciliation , the rgw pod does not give any errors but Rook-operator says unknown error. So do you remove the bucket manually from the backend after applying finaliser on OBC?

**Comment by zerkms** [comment]
> So do you remove the bucket manually from the backend after applying finaliser on OBC?

No, when I checked the bucket manually using rgw cli the bucket has already been removed.

After that I removed OBC finaliser, and also removed the rgw user (that is created by default for every user).

**Comment by thotz** [comment]
Oh I see so may be this issue related to https://github.com/rook/rook/issues/9494

**Comment by zerkms** [comment]
I have not re-verified it recently, but unless it was fixed by accident - it should still be there?

**Comment by BlaineEXE** [comment]
@zerkms please re-verify. I believe this was fixed by #9603

**Comment by zerkms** [comment]
@BlaineEXE right, thanks! Let's put it different way then: I will create a task in an internal bug tracker to re-verify it later when I have spare time for that, and if it's still reproducible on a later version - I will reopen this report. Until then - I'm closing it here.

Thanks again! :-)
